### DISCUSSION TITLE
Config file for exiting insert mode

### DISCUSSION THREAD

**Discussion by bytesizelearning** [mickael-menu/ShadowVim#52] [Ideas]
Now that Xcode handles insert mode it is no longer possible to map a key sequence to exit insert mode. Personally this was the only `imap` mapping I used, so it would be great to get some sort of configuration file maybe in json similar to how Sublime handles it.

```json
{ "keys": ["j", "j"], "command": "exit_insert_mode" }
```
Would love to hear your opinions on it.

**Comment by mickael-menu** [comment]
Agreed, this is such a common use case that it might be worth being implemented in ShadowVim.

Ideally, we could support arbitrary mappings and actions. But a first step could be just an option to exit insert mode when a set of letters is pressed.

If someone wants to take a stab at this, you should be able here to accumulate the key events until it matches `jj`, and if not send them to Neovim with `.nvimInput(...)`.

https://github.com/mickael-menu/ShadowVim/blob/6b6e5de64573a3c1f4d24d17b78cfff1899c0c0f/Sources/Mediator/Buffer/CooperativeBufferState.swift#L287
